src/battle_ai_main.c

```diff
@@ -2059,12 +2058,8 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
               || DoesPartnerHaveSameMoveEffect(BATTLE_PARTNER(battlerAtk), battlerDef, move, aiData->partnerMove))
                 ADJUST_SCORE(-9);
             break;
-        case EFFECT_BURN_UP:
-            if (!IS_BATTLER_OF_TYPE(battlerAtk, TYPE_FIRE))
-                ADJUST_SCORE(-10);
-            break;
-        case EFFECT_DOUBLE_SHOCK:
-            if (!IS_BATTLER_OF_TYPE(battlerAtk, TYPE_ELECTRIC))
+        case EFFECT_FAIL_IF_NOT_ARG_TYPE:
+            if (!IS_BATTLER_OF_TYPE(battlerAtk, gBattleMoves[move].argument))
                 ADJUST_SCORE(-10);
```

Comment:
I will test this but it doesn't need special handling. The AI will recognize that it can't use the move

Reply:
OK, but then I'd rather leave that in until it's tested